src/lib/notcurses.c

```diff
@@ -593,6 +593,10 @@ inline int ncplane_cursor_move_yx(ncplane* n, int y, int x){
   return 0;
 }
 
+inline int ncplane_cursor_move_rel(ncplane* n, int y, int x){
+  return ncplane_cursor_move_yx(n, n->y + y, n->x + x);
```

Comment:
this is no bueno. it works for relative movement of *planes*, but not for the *cursor*, because `ncplane_cursor_move_yx()` interprets -1 as "hold constant" rather than "error". so you need to check for that.

Reply:
Wow, that's embarrassing. I literally propose that feature and forgot about it at the same day.

So, what should be done when a user tries to move cursor outside the ncplane?
- Ignore the command, return -1 (the baseline)
- Truncate movement to the edges of ncplane and stop there,
- Wrap around x axis, but do nothing if outside horizontal bounds.

The first option sounds like the most "compatible" with absolute version, but the third offers some more functionality.

Reply:
Would be far easier to support both, if there were optional arguments in C. :/

Reply:
it should act like `ncplane_move_cursor_yx()` in all details, and in fact ought be a wrapper around `ncplane_move_cursor_yx()` in my opinion. oh i see why you're asking, because you need to know what to do with a relative move that goes negative. please do the following:

```c
if(x < 0){
  logerror("Negative target x %d\n", x);
}
```

with y in place of x for relative move up. you don't need to check anything for relmove down or right, only up and left.

i've updated some documentation, which was full of lies.

Reply:
Well, that's exactly what I did. I let the _xy handle invalid targets, and -1 targets explicitly.